alg/teca_damper.cxx

```diff
@@ -0,0 +1,217 @@
+
+    // get the name of the array to request
+    std::vector<std::string> damper_vars = this->get_damper_variables(request);
+    if (damper_vars.empty())
+    {
+        TECA_ERROR("damper variables were not specified")
+        return up_reqs;
+    }
+
+    // pass the incoming request upstream, and
+    // add in what we need
+    teca_metadata req(request);
+    std::set<std::string> arrays;
+    if (req.has("arrays"))
+        req.get("arrays", arrays);
+    arrays.insert(damper_vars.begin(), damper_vars.end());
+
+    req.insert("arrays", arrays);
+
+    // send up
+    up_reqs.push_back(req);
+
+    return up_reqs;
+}
+
+// --------------------------------------------------------------------------
+const_p_teca_dataset teca_damper::execute(
+    unsigned int port,
+    const std::vector<const_p_teca_dataset> &input_data,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id()
+        << "teca_damper::execute" << endl;
+#endif
+    (void)port;
+
+    // get the input
+    const_p_teca_cartesian_mesh in_mesh =
+        std::dynamic_pointer_cast<const teca_cartesian_mesh>(
+            input_data[0]);
+    if (!in_mesh)
+    {
+        TECA_ERROR("empty input, or not a mesh")
+        return nullptr;
+    }
+
+    // create output and copy metadata, coordinates, etc
+    p_teca_cartesian_mesh out_mesh =
+        std::dynamic_pointer_cast<teca_cartesian_mesh>(in_mesh->new_instance());
+
+    out_mesh->shallow_copy(
+        std::const_pointer_cast<teca_cartesian_mesh>(in_mesh));
+
+    // get the input array names
+    std::vector<std::string> damper_vars = this->get_damper_variables(request);
+    if (damper_vars.empty())
+    {
+        TECA_ERROR("A damper variable was not specified")
+        return nullptr;
+    }
+
+    // get threshold values
+    double filter_lat_width_val = this->filter_lat_width_value;
+
+    if (std::isnan(filter_lat_width_val)
+        && request.has("teca_damper::filter_lat_width_value"))
+        request.get("teca_damper::filter_lat_width_value", filter_lat_width_val);
+
+    if (std::isnan(filter_lat_width_val))
+    {
+        TECA_ERROR("A filter_lat_width (delta-Y) value was not specified")
+        return nullptr;
+    }
+
+    // get the coordinate axes
+    const_p_teca_variant_array yc = in_mesh->get_y_coordinates();
+    const_p_teca_variant_array xc = in_mesh->get_x_coordinates();
+
+    size_t n_lat = yc->size();
+    size_t n_lon = xc->size();
+
+    p_teca_variant_array filter_array = yc->new_instance(n_lat);
+
+    // Get the gaussian filter
+    NESTED_TEMPLATE_DISPATCH_FP(teca_variant_array_impl,
+        filter_array.get(),
+        _COORD,
+
+        const NT_COORD *p_yc = static_cast<const TT_COORD*>(yc.get())->get();
+
+        NT_COORD *p_filter_array = static_cast<TT_COORD*>(filter_array.get())->get();
+
+        ::get_lat_filter(p_filter_array, p_yc, n_lat, filter_lat_width_val);
+
+        size_t n_arrays = damper_vars.size();
```

Comment:
we should accept a list of variables to damp, rather than damping them all

Reply:
I thought specifying `TECA_ALGORITHM_VECTOR_PROPERTY(std::string, damper_variable)` vector of variables is a sort of choosing them. Please let me know if that's not the case. If not please share an example algorithm where variables are specified.

Reply:
You are right!, about this my bad.